bin/detect_ips.sh

```diff
@@ -1,48 +1,16 @@
 #!/bin/bash
 
-PUBLIC_IP=`dig +short myip.opendns.com @resolver1.opendns.com`
-## Fallback to a different dns provider
+# Fetch IPv4 WAN address using ifconfig.me, fallback to ipinfo.io
+PUBLIC_IP=$(curl -4 -s https://ifconfig.me)
 if [ -z "$PUBLIC_IP" ]; then
-    PUBLIC_IP=`dig -4 TXT +short o-o.myaddr.l.google.com @ns1.google.com | sed 's|"||g'`
+    PUBLIC_IP=$(curl -4 -s https://ipinfo.io/ip)
 fi
 
-runOnMac=false
-int2ip() { printf ${2+-v} $2 "%d.%d.%d.%d" \
-        $(($1>>24)) $(($1>>16&255)) $(($1>>8&255)) $(($1&255)) ;}
-ip2int() { local _a=(${1//./ }) ; printf ${2+-v} $2 "%u" $(( _a<<24 |
-                  ${_a[1]} << 16 | ${_a[2]} << 8 | ${_a[3]} )) ;}
-while IFS=$' :\t\r\n' read a b c d; do
-    [ "$a" = "usage" ] && [ "$b" = "route" ] && runOnMac=true
-    if $runOnMac ;then
-        case $a in
-            gateway )    gWay=$b  ;;
-            interface )  iFace=$b ;;
-        esac
-    else
-        [ "$a" = "0.0.0.0" ] && [ "$c" = "$a" ] && iFace=${d##* } gWay=$b
-    fi
-done < <(/sbin/route -n 2>&1 || /sbin/route -n get 0.0.0.0/0)
-ip2int $gWay gw
-while read lhs rhs; do
-    [ "$lhs" ] && {
-        [ -z "${lhs#*:}" ] && iface=${lhs%:}
-        [ "$lhs" = "inet" ] && [ "$iface" = "$iFace" ] && {
-            mask=${rhs#*netmask }
-            mask=${mask%% *}
-            [ "$mask" ] && [ -z "${mask%0x*}" ] &&
-                printf -v mask %u $mask ||
-                ip2int $mask mask
-            ip2int ${rhs%% *} ip
-            (( ( ip & mask ) == ( gw & mask ) )) &&
-                int2ip $ip myIp && int2ip $mask netMask
-        }
-    }
-done < <(/sbin/ifconfig)
-
-echo "KADCAST_PUBLIC_ADDRESS=$PUBLIC_IP:9000"
-if [ -z "$myIp" ]; then
-    echo "KADCAST_LISTEN_ADDRESS=$PUBLIC_IP:9000"
-else
-    echo "KADCAST_LISTEN_ADDRESS=$myIp:9000"
```

Comment:
I'm not sure we should remove the private ip detection too

Reply:
@herr-seppia you're right! I was being too enthusiastic. This _should_ behave the same as it did before and take the local IP into account again. Tested it on a local machine.

In `curl` I specify the SSL version. Without doing so, it fails on Hetzner for me. DO and local machine seem to be fine without it though.